pymc_marketing/clv/utils.py

```diff
@@ -597,3 +598,185 @@ def rfm_train_test_split(
     train_test_rfm_data["test_T"] = time_delta / time_scaler  # type: ignore
 
     return train_test_rfm_data
+
+
+def rfm_segments(
+    transactions: pd.DataFrame,
+    customer_id_col: str,
+    datetime_col: str,
+    monetary_value_col: str,
+    segment_config: dict | None = None,
+    observation_period_end: str | pd.Period | datetime | None | None = None,
```

Comment:
There are two `None`s :)

Reply:
Also below in all the remaining function arguments ;)

Reply:
Whoops; updating the minimum python version to 3.10 seems to have changed the type hinting syntax. Will get this sorted out.